### DISCUSSION TITLE
Blocking issue with sending transactional emails via Supabase

### DISCUSSION THREAD

**Discussion by HiThereWsp** [supabase/supabase#31134] [Contribute to Supabase]
Hi there,

I’m facing a blocking issue with sending transactional emails through Supabase.

What I’ve tried:
Resend:
Integration with Supabase.
Configuration via SMTP.
Amazon SES (sandbox):
Configuration via SMTP.
In both cases, I tested sending emails to addresses stored in my Supabase database, but none were delivered.

Why this is blocking:
I can’t send any emails—neither transactional nor marketing—which is preventing me from engaging with my users and offering a proper experience.

Do you have any insights or examples of a working configuration? Any help would be greatly appreciated.

Thanks in advance!

Best regards,
HiThereWassup

**Comment by GaryAustin1** [comment]
Not commenting on marketing emails as nothing from Supabase deals with that.  You would have to use your provider API's.

Both Resend and SES are used by many Supabase users without issue.  

You'll need to look into auth error logs in the dashboard to see if anything is being returned.  Most SMTP providers require you to confirm you own a website address.

The only thing needed on the Supabase side is setting up your SMTP credentials.

**Comment by HiThereWsp** [comment]
Thanks for the reply!

I’m sorry, but I’m not a developer. I’ve already confirmed my domain with both providers (Resend and Amazon SES), and the log doesn't show any error messages, as you can see in the attached screenshot.

The logs only show successful token requests and logins, but there are no errors related to the email sending process. I've also made sure that the SMTP credentials are correctly set up on the Supabase side.

Could you please help me identify what might be causing the issue or if there's anything else I should check?

Thanks again!

**Reply by GaryAustin1** [reply]
Not sure what screen shots you refer to.
How are you testing sending emails?
If you use the dashboard to send a password reset on an existing user does it go thru?
Can you invite a new user from the dashboard?

If you are making Auth API calls maybe show the calls you are making?  Is the email provider setup in the dashboard authentication section set to confirm emails if you are doing signups?

Have you changed the default templates?

I would debug with Resend as many users are using that.

**Comment by HiThereWsp** [comment]
Thank you for your suggestions. I have reviewed the options you mentioned and I also found the following logs, which could be relevant to the issue:

sql
Copier le code
Event Message
{"auth_event":{"action":"user_recovery_requested","actor_id":"2bee9cc4-d2a1-49ff-9d8c-a7bc3c2057b6","actor_username":"mymailadressfortest@gmail.com","actor_via_sso":false,"log_type":"user"},"component":"api","error":"535 Authentication Credentials Invalid","level":"error","method":"POST","msg":"500: Error sending recovery email","path":"/recover","referer":"https://mywebapp
<img width="950" alt="Capture d’écran 2024-12-13 à 20 56 11" src="https://github.com/user-attachments/assets/78437c46-b209-4ad7-b476-2eef60fd8714" />
/reset-password","remote_addr":"2.56.190.175","request_id":"8f12c67966c645f3-DFW","time":"2024-12-13T03:12:58Z"}
Timestamp
2024-12-13T03:12:58.000Z
I plan to analyze these options further tomorrow morning and will keep you updated.

Thanks again for your help!

**Reply by GaryAustin1** [reply]
It looks like your SMTP settings in the dashboard do not match the correct ones for your provider.   I'll check back tomorrow.

**Reply by HiThereWsp** [reply]
Hi Gary,

I apologize for the delayed response and hope you’re still available to help me with this issue.

I’ve just tested sending an invitation email via the Supabase dashboard, and it was successfully sent. However, when I retried the registration process, no email was sent. I’ve attempted multiple configurations, but none have worked so far. Here’s the latest setup I’ve tried:

New user registration → Triggers on_auth_user_created.

Insert into welcome_emails table → Calls the handle_new_user() function.

Email sending via Resend → Handled by the Edge Function process-welcome-emails.

Despite this setup, the emails are still not being sent. Could you please advise on what I might be missing or what steps I should take next?

Thank you so much for your time and support!

Best regards,

**Reply by GaryAustin1** [reply]
It is is confusing as to how you are sending emails as you also have an edge function involved with an email.
Are you using the SMTP logic built into Supabase to send your confirmation email and then an extra email?
Are you getting any errors in the Postgres logs?  Auth logs?   
Originally you were talking about a reset email.

It might be easier to go to Supabase Discord for back and forth with me or other users than here.

**Reply by GaryAustin1** [reply]
Check the Postgres log to see if the error is coming from your auth.users trigger as that might have user_name in it.

**Comment by HiThereWsp** [comment]
Hi Gary,

Thank you for your response. Let me clarify my setup and the issue I'm facing:

SMTP Configuration : I am using the built-in Supabase x Resend integration.

Edge Function : I have a custom Edge Function (process-welcome-emails) to send a welcome email when a new user registers.

Current Issue:
Initially, the issue was with sending welcome emails via the Edge Function. However, I noticed that when I try to send a password reset email or magic link via the Supabase dashboard, I get the error:
"Failed to send magic link: property first_name should not exist".

I have checked all email templates in Authentication → Email Templates, and there is no reference to first_name in any of them.

There is also no button to reset the templates to their default values.

What I've Tried:
Verified that the SMTP credentials are correct and that my domain is confirmed with Resend.

Checked the Auth logs and found no errors except the one mentioned above.

Tested sending a password reset email to a new user with minimal metadata (only email and password), but the issue persists.

Questions:
Could the first_name property be coming from somewhere else (e.g., user metadata or a default template that I can't see)?

Is there a way to reset the email templates to their default values if the button is missing?

Do you think I should remove the Edge Functions and try again, or is there another approach you recommend?

I am quite confused and unsure of what to do next. Any guidance would be greatly appreciated.

Thank you in advance!

I will also ask it on Discord

**Comment by HiThereWsp** [comment]
<img width="581" alt="Capture d’écran 2025-01-17 à 18 09 36" src="https://github.com/user-attachments/assets/12de234b-1978-4614-ba9d-bd54aa2dec31" />
[image: image.png]

Can you see something related here ?

Le ven. 17 janv. 2025 à 17:57, GaryAustin1 ***@***.***> a
écrit :

> Check the Postgres log to see if the error is coming from your auth.users
> trigger as that might have user_name in it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/supabase/discussions/31134#discussioncomment-11871848>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BLYKG26IXTKXBACCSCA3RTD2LF4GBAVCNFSM6AAAAABTS7CKOGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCOBXGE4DIOA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
*Andy GUITTEAUD *
